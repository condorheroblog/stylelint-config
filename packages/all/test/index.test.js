import { readFileSync } from "node:fs";
import { join } from "node:path";
import stylelint from "stylelint";
import { beforeEach, describe, expect, test } from "vitest";

import basicConfig from "../";

const config = { ...basicConfig, ignoreFiles: [] };
const validateCssPath = join(__dirname, "./index.css");
const validateScssPath = join(__dirname, "./index.scss");
const validateJssPath = join(__dirname, "./index.tsx");
const validateVuePath = join(__dirname, "./index.vue");
const validateCss = readFileSync(validateCssPath, "utf-8");
const validateScss = readFileSync(validateScssPath, "utf-8");
const validateJss = readFileSync(validateJssPath, "utf-8");

describe("validate CSS", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: validateCssPath,
      configBasedir: __dirname,
    });
  });

  test("CSS did error", () => {
    return result.then(data => expect(data.errored).toBeTruthy());
  });

  test("CSS had 3 warning", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(1));
  });
});

describe("validate SCSS", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: validateScssPath,
      configBasedir: __dirname,
    });
  });

  test("SCSS did error", () => {
    return result.then(data => expect(data.errored).toBeTruthy());
  });

  test("SCSS had 1 warning", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(1));
  });

  test("SCSS rule flagged", () => {
    return result.then(data =>
      expect(data.results[0].warnings[0].rule).toBe(
        "scss/dollar-variable-colon-space-after",
      ),
    );
  });
});

describe("validate css-in-js", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: validateJssPath,
      configBasedir: __dirname,
    });
  });

  test("css-in-js did error", () => {
    return result.then(data => expect(data.errored).toBeTruthy());
  });

  test("css-in-js had 1 warning", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(1));
  });
});

describe("validate vue", () => {
  let result;

  beforeEach(() => {
    result = stylelint.lint({
      config,
      files: validateVuePath,
      configBasedir: __dirname,
    });
  });

  test("vue did error", () => {
    return result.then(data => expect(data.errored).toBeTruthy());
  });

  test("vue had 1 warning", () => {
    return result.then(data => expect(data.results[0].warnings).toHaveLength(2));
  });
});

describe("validate file snapshot", () => {
  test("index.css", () => {
    return expect(validateCss).toMatchSnapshot();
  });

  test("index.scss", () => {
    return expect(validateScss).toMatchSnapshot();
  });

  test("index.tsx", () => {
    return expect(validateJss).toMatchSnapshot();
  });

  test("index.vue", () => {
    return expect(validateVuePath).toMatchSnapshot();
  });
});
