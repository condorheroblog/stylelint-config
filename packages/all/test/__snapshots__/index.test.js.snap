// Vitest Snapshot v1

exports[`validate file snapshot > index.css 1`] = `
"div {
  position:relative;
}
"
`;

exports[`validate file snapshot > index.scss 1`] = `
"$w10:10px;
"
`;

exports[`validate file snapshot > index.tsx 1`] = `
"import styled from \\"@emotion/styled\\";

const Title = styled.h2\`margin: 0 0 1rem;
font-size: 1.5rem;
width: calc(5% - 10em);
\`;

function Index() {
  return (
    <>
      <Title>Learn React</Title>
    </>
  );
}

export default Index;
"
`;

exports[`validate file snapshot > index.vue 1`] = `
"<style lang=\\"scss\\">
//comment
</style>
<style>
:global(.foo),
::v-deep(.foo),
::v-slotted(.foo),
::v-global(.foo),
:deep(.foo),
:slotted(.foo) {
    color: red;
}
</style>
"
`;
